### PR TITLE
[Feature] Dev 1.2.1 linkis metadata query add database

### LINKED ISSUE #2443: [Feature] linkis metadata query add database
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

Extend the database supported by Linkis DataSource, extend dm, greenplum, kingbase, oracle, postgres, sqlserver

### Description

metadataManager Service implementation
Based on the mysql module, it integrates dm, greenplum, kingbase, oracle, postgres, sqlserver, and the protocol is separated from sql. The metadatamanager reflects multiple data sources, and is extended based on the mysql module and merged into one module.

### Use case

1.linkis-metadata-query-server
loader：MetaClassLoaderManager maps mysql collection relationship classes to contain a variety of relational databases
2.linkis-metadata-query-service-mysql
db2,dm, greenplum, kingbase, oracle, postgres, sqlserver implement their own
Introduce each driver separately

dm driver : Dm7JdbcDriver16-7.6.0.165.jar
greenplum driver:greenplum-jdbc-5.1.4
kingbase driver:kingbase8-8.2.0.jar
oracle driver :ojdbc6-11.2.0.3.jar
postgres driver:postgresql-42.3.1.jar
sqlserver driver:
msbase-1.0.jar
mssqlserver-1.0.jar
msutil-1.0.jar
db2 driver:db2jcc4.jar

_No response_

### solutions

metadataManager Service implementation
Based on the mysql module, it integrates dm, greenplum, kingbase, oracle, postgres, sqlserver, and the protocol is separated from sql. The metadatamanager reflects multiple data sources, and is extended based on the mysql module and merged into one module.

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Dlimeng**
### What is the purpose of the change
Extend the database supported by Linkis DataSource, extend dm, greenplum, kingbase, oracle, postgres, sqlserver
Fixes  #2443

### Brief change log
metadataManager Service implementation
Based on the mysql module, it integrates dm, greenplum, kingbase, oracle, postgres, sqlserver, and the protocol is separated from sql. The metadatamanager reflects multiple data sources, and is extended based on the mysql module and merged into one module.

### Does this pull request potentially affect one of the following parts:

1.linkis-metadata-query-server
loader：MetaClassLoaderManager maps mysql collection relationship classes to contain a variety of relational databases
2.linkis-metadata-query-service-mysql
db2,dm, greenplum, kingbase, oracle, postgres, sqlserver implement their own
Introduce each driver separately

dm driver : Dm7JdbcDriver16-7.6.0.165.jar
greenplum driver:greenplum-jdbc-5.1.4
kingbase driver:kingbase8-8.2.0.jar
oracle driver :ojdbc6-11.2.0.3.jar
postgres driver:postgresql-42.3.1.jar
sqlserver driver:
msbase-1.0.jar
mssqlserver-1.0.jar
msutil-1.0.jar
db2 driver:db2jcc4.jar

### CODE REVIEW

File: linkis-dist/package/db/linkis_dml.sql [hunk 1]
```diff
@@ -465,6 +472,67 @@ INSERT INTO `linkis_ps_dm_datasource_type_key` (`data_source_type_id`, `key`, `n
...
+INSERT INTO `linkis_ps_dm_datasource_type_key` (`data_source_type_id`, `key`, `name`, `default_value`, `value_type`, `scope`, `require`, `description`, `value_regex`, `ref_id`, `ref_value`, `data_source`, `update_time`, `create_time`) VALUES (12, 'username', '用户名(Username)', NULL, 'TEXT', NULL, 1, '用户名(Username)', '^[0-9A-Za-z_-]+$', NULL, NULL, NULL,  now(), now());
+INSERT INTO `linkis_ps_dm_datasource_type_key` (`data_source_type_id`, `key`, `name`, `default_value`, `value_type`, `scope`, `require`, `description`, `value_regex`, `ref_id`, `ref_value`, `data_source`, `update_time`, `create_time`) VALUES (12, 'password', '密码(Password)', NULL, 'PASSWORD', NULL, 1, '密码(Password)', '', NULL, NULL, NULL,  now(), now());
+INSERT INTO `linkis_ps_dm_datasource_type_key` (`data_source_type_id`, `key`, `name`, `default_value`, `value_type`, `scope`, `require`, `description`, `value_regex`, `ref_id`, `ref_value`, `data_source`, `update_time`, `create_time`) VALUES (12, 'instance', '实例名(instance)', NULL, 'TEXT', NULL, 1, '实例名(instance)', NULL, NULL, NULL, NULL,  now(), now());
```
Comment (casionone) [inline linkis-dist/package/db/linkis_dml.sql:533]: The Incremental SQL statements also need to be added to linkis-dist/package/db/upgrade/1.2.1_schema

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/server/src/main/java/org/apache/linkis/metadata/query/server/loader/MetaClassLoaderManager.java [hunk 2]
```diff
@@ -78,6 +85,7 @@ public BiFunction<String, Object[], Object> getInvoker(String dsType) throws Err
         }
         if (needToLoad) {
             MetaServiceInstance finalServiceInstance1 = serviceInstance;
+            String finalBaseType = baseType;
```
Comment (casionone) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/server/src/main/java/org/apache/linkis/metadata/query/server/loader/MetaClassLoaderManager.java:88]:
```
String finalBaseType = baseType;
String componentLib = stdLib + "/" + finalBaseType;
```
`String finalBaseType = baseType` not necessary?

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/src/main/java/org/apache/linkis/metadata/query/service/db2/SqlConnection.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        Statement stmt = null;
+        ResultSet rs = null;
+        try {
+            stmt = conn.createStatement();
+            rs = stmt.executeQuery("list database directory");
+            // rs = stmt.executeQuery("SELECT * FROM SYSIBMADM.APPLICATIONS WITH UR");
+            // rs = stmt.executeQuery("select * from syscat.tables");
+            while (rs.next()) {
+                String db_name = rs.getString("DB_NAME");
+                System.out.println(db_name);
```
Comment (casionone) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/src/main/java/org/apache/linkis/metadata/query/service/db2/SqlConnection.java:81]: " System.out.println" need to be removed

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/src/main/java/org/apache/linkis/metadata/query/service/db2/SqlConnection.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        connectMessage = new ConnectMessage(host, port, username, password, extraParams);
+        conn = getDBConnection(connectMessage, database);
+        // Try to create statement
+        Statement statement = conn.createStatement();
+        statement.close();
+    }
+
+    public List<String> getAllDatabases() throws SQLException {
+        // db2 "select schemaname from syscat.schemata"
+        List<java.lang.String> dataBaseName = new ArrayList<>();
```
Comment (casionone) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/src/main/java/org/apache/linkis/metadata/query/service/db2/SqlConnection.java:71]: optimize code style `List<String>`

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/pom.xml [hunk 5]
```diff
@@ -55,11 +56,69 @@
...
+<!--            <scope>system</scope>-->
+<!--            <systemPath>${basedir}/src/main/lib/msutil-1.0.jar</systemPath>-->
+<!--        </dependency>-->
+<!--        <dependency>-->
+<!--            <groupId>com.microsoft.jdbc.sqlserver</groupId>-->
+<!--            <artifactId>mssqlserver</artifactId>-->
+<!--            <version>1.0</version>-->
+<!--            <scope>system</scope>-->
+<!--            <systemPath>${basedir}/src/main/lib/mssqlserver-1.0.jar</systemPath>-->
+<!--        </dependency>-->
```
[the remark this answers is not in the document]
Comment (Dlimeng) [reply linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/mysql/pom.xml:121]: postgresql can be referenced directly, other references need to be downloaded from the official website

File: linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/domain/MetaColumnInfo.java [hunk 6]
```diff
@@ -30,6 +30,24 @@ public class MetaColumnInfo implements Serializable {
...
+        this.colComment = colComment;
+    }
+
+    public int getLength() {
+        return length;
+    }
+
+    public void setLength(int length) {
+        this.length = length;
+    }
```
Comment (casionone) [inline linkis-public-enhancements/linkis-datasource/linkis-metadata-query/common/src/main/java/org/apache/linkis/metadata/query/common/domain/MetaColumnInfo.java:50]: After this DTO class MetaColumnInfo is changed, the compatibility with the client needs to be considered for class `org.apache.linkis.datasource.client.response.MetadataGetColumnsResult`, whether it is better to provide a unit test to verify the compatibility, specify the linkis-datasource-client that depends on version 1.1.2.

File: docs/info-1.2.1.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+## 参数变化 
+
+
+| 模块名(服务名)| 类型  |     参数名                                                | 默认值             | 描述                                                    |
+| ----------- | ----- | -------------------------------------------------------- | ---------------- | ------------------------------------------------------- |
+|ps-metadataquery | 新增  |        | linkis-metadata-query-service-mysql| 在mysql元数据服务兼容oracle,kingbase,postgresql,sqlserver,db2,greenplum,dm,驱动外部引入 |
```
Comment (casionone) [inline docs/info-1.2.1.md:6]: 参数名  wds.linkis.server.mdq.mysql.relationship

**Review by casionone — APPROVED** [review]

**Pull request merged**